src/Views/Profile/Profile.js

```diff
@@ -1,25 +1,97 @@
 // @flow
-import React, {Component} from 'react';
-import {Paper, RaisedButton} from "material-ui";
-import {PropTypes} from 'prop-types';
+import React, { Component } from 'react';
+import { Paper, RaisedButton, Avatar, FlatButton, TextField } from "material-ui";
+import { PropTypes } from 'prop-types';
+import Messages from "../../Messages";
 
 export default class Profile extends Component {
 
+    constructor(props) {
+        super(props);
+        this.handleEditClick = this.handleEditClick.bind(this);
+        this.handleSaveClick = this.handleSaveClick.bind(this);
+        this.state = { isEditable: false };
+    }
+
     static propTypes = {
         handleLogout: PropTypes.func.isRequired,
-        profile: PropTypes.object.isRequired
+        handleUpdateProfile: PropTypes.func.isRequired,
+        profile: PropTypes.object.isRequired,
     };
 
+    handleEditClick() {
+        this.setState({ isEditable: true });
+    }
+
+    handleSaveClick() {
+        this.setState({ isEditable: false });
+        this.props.handleUpdateProfile(
+            this.refs.textFieldPosition.getValue(),
+            this.refs.textFieldCompanyName.getValue(),
+            this.refs.textFieldPhoneNumber.getValue()
+        );
+    }
+
     render() {
-        const {profile} = this.props;
+        const { profile } = this.props;
+        const isEditable = this.state.isEditable;
+
+        let editButton = null;
+        if (isEditable) {
+            // assign editable profile view
+            editButton = <FlatButton label={Messages.save} style={{ position: 'absolute', right: 0, marginTop: 15 }} onClick={this.handleSaveClick} />;
+        } else {
+            // assign static profile view
+            editButton = <FlatButton label={Messages.edit} style={{ position: 'absolute', right: 0, marginTop: 15 }} onClick={this.handleEditClick} />
+        }
+
         return (
             <div>
-                <Paper style={{padding: 30, margin: 30, textAlign: 'center'}} zDepth={1}>
-                    <h1>{profile.displayName}</h1>
-                    <h4>{profile.email}</h4>
-                    <img style={{width: 200, padding: 30}} src={profile.photoURL} alt="profile"/>
-                    <RaisedButton style={{width: '90%'}} label="Wyloguj się" secondary={true} onTouchTap={() => this.props.handleLogout()}/>
-                </Paper>
+                {editButton}
+                <div style={{ paddingTop: 51, textAlign: 'center' }}>
+                    <Avatar src={profile.photoURL} size={160} /><br />
```

Comment:
\<br/\> to break line, będąc w \<div\> (komponent blokowy) a nie węzeł tekstu (jak np. \<p\>) nie masz "line" do łamania - więc to jest bład :) wystarczy tu margines/padding, mam nadzieje, ze uda sie obejść bez \<br /\> :)